README.md

```diff
@@ -100,9 +123,87 @@ act_when($isChuckNorrisMighty, new ChuckNorris($data));
 act_unless($isChuckNorrisPuny, new ChuckNorris($data));
 ```
 
+### Dependency Injection
+
+You can even injection actions as a dependency with your app!
+
+```php
+use Kirschbaum\Actions\Contracts\Actionable;
+
+public function index (Actionable $action)
+{
+    $action->act(new ChuckNorris($data));
+    $action->actWhen($isChuckNorrisMighty, new ChuckNorris($data));
+    $action->actUnless($isChuckNorrisPuny, new ChuckNorris($data));
+}
+```
+
+## Handling Failures
+
+We all know Chuck Norris isn't going to fail us, but he isn't the only one using this... Handling failures is pretty easy with actions. Out of the box, any exceptions get handled by Laravel's exception handler. If you'd rather implement your own logic during a failure, add a `failed()` method to your action. It's that easy!
+
+```php
+/**
+ * Handle failure of the action.
+ *
+ * @throws Throwable
+ *
+ * @return mixed
+ */
+public function failed(Throwable $exception)
+{
+    event(new VanDammeFailedEvent);
+}
+```
+
+Hashtag #BAM!
+
+### Custom Exceptions
+
+Another option for handling failures is to tell the action to fire its own exception. If you don't need the extra overhead of writing your own `failed()` method, you can just tell your action to throw a custom exception. It's as simple as just defining the exception you want thrown from the action.
```

Comment:
```suggestion
Another option for handling failures is to tell the action to throw its own exception. If you don't need the extra overhead of writing your own `failed()` method, you can just tell your action to throw a custom exception. It's as simple as just defining the exception you want thrown from the action.
```

Is _throw_ more correct than "fire" in this context? I'm actually not sure. Maybe it's not. Now I'm second-guessing myself.